src/main/scala/com/thoughtworks/enableIf.scala

```diff
@@ -3,8 +3,48 @@ package com.thoughtworks
 import scala.annotation.StaticAnnotation
 import scala.reflect.internal.annotations.compileTimeOnly
 import scala.reflect.macros.Context
+import scala.util.matching.Regex
+
 
 object enableIf {
+  private def getRegex(artifactId: String, regex: Regex): Regex = {
+    new Regex(s".*${artifactId}-${regex.toString}")
+  }
+
+  private def getRegex(artifactId: String, version: String): Regex = {
+    val versionRegex = s"${version.replace(".", "\\.")}.*"
+    getRegex(artifactId, new Regex(versionRegex))
+  }
+
```

Comment:
```suggestion
```

Use capture group instead of building a regular expression from string concatenation

Reply:
Using regex groups now! Please review

Reply:
I mean you might not need `getRegex`. Instead, you can extract artifactId and version with the help with a constant regex, and then match the version with another regex.

Reply:
got it. The constant regex approach is much better!